src/sentry/sentry_metrics/indexer/cache.py

```diff
@@ -11,9 +11,10 @@
 
 
 class StringIndexerCache:
-    def __init__(self, version: int, cache_name: str):
+    def __init__(self, version: int, cache_name: str, partition_key: Optional[str] = None):
```

Comment:
Looking at this again, I don't really get why we need this at all. Why not just have the caller pass in a different cache_name for the two variants.

Reply:
@lynnagara do you mean change the `cache_name` for both? since right now it's `default`? I think that would mean basically adding a new backend like it does here 

https://github.com/getsentry/sentry/blob/458d01c46a8338c544def0ce01eea9d5ab88fc62/src/sentry/utils/pytest/sentry.py#L129-L132

I thought we just wanted a way to easily distinguish between the keys

Reply:
Oh i thought `cache_name` was the part that was `indexer`. I see that is hardcoded now.